docs/content/guides/azure.md

```diff
@@ -0,0 +1,158 @@
+
+# etcd
+az network nsg rule create -g $GROUP --nsg-name talos -n etcd --priority 1003 --destination-port-ranges 2379-2380 --direction inbound
+
+# Proxyd -> Kubernetes API Server
+az network nsg rule create -g $GROUP --nsg-name talos -n kube --priority 1004 --destination-port-ranges 6443 --direction inbound
+```
+
+## Cluster Configuration
+
+After getting the network security group set up, we'll need to allocate public
+IPs for our master nodes. The example below assumes a HA control plane of 3 nodes.
+You can adjust this for your needs.
+
+```bash
+# Reserve public IPs
+az network public-ip create -g $GROUP --name talos-master-1 --allocation-method static
+az network public-ip create -g $GROUP --name talos-master-2 --allocation-method static
+az network public-ip create -g $GROUP --name talos-master-3 --allocation-method static
+
+# Gather public IPs into a comma separated string
+MASTERIPS=$(az network public-ip list -g $GROUP -o tsv --query [].ipAddress | paste -sd,)
+
+# Generate a default Talos config for this cluster
+# # This should generate master-{1,2,3}.yaml, worker.yaml, and talosconfig in your PWD
+./osctl config generate cluster.local $MASTERIPS
+```
+
+## Compute Creation
+
+```bash
+# Create master nodes
+# # `--admin-username` and `--generate-ssh-keys` are required by the az cli,
+# # but are not actually used by talos
+# # `--os-disk-size-gb` is the backing disk for Kubernetes and any workload containers
+# # `--boot-diagnostics-storage` is to enable console output which may be necessary
+# # for troubleshooting
+az vm create \
+  --name talos1 \
+  --image talos \
+  --custom-data ./master-1.yaml \
+  --public-ip-address talos-master-1 \
+  -g $GROUP \
+  --admin-username talos \
+  --generate-ssh-keys \
+  --verbose \
+  --boot-diagnostics-storage $STORAGE_ACCOUNT \
+  --nsg talos \
+  --os-disk-size-gb 64 \
+  --no-wait
+az vm create --name talos2 --image talos --custom-data ./master-2.yaml  --public-ip-address talos-master-2 -g $GROUP --admin-username talos --generate-ssh-keys --verbose --boot-diagnostics-storage $STORAGE_ACCOUNT --nsg talos --os-disk-size-gb 64 --no-wait
+az vm create --name talos3 --image talos --custom-data ./master-3.yaml  --public-ip-address talos-master-3 -g $GROUP --admin-username talos --generate-ssh-keys --verbose --boot-diagnostics-storage $STORAGE_ACCOUNT --nsg talos --os-disk-size-gb 64 --no-wait
+
+## Create worker nodes, reuse as needed
+az vm create --name talos4 --image talos --custom-data ./worker.yaml -g $GROUP --admin-username talos --generate-ssh-keys --verbose --boot-diagnostics-storage $STORAGE_ACCOUNT --nsg talos --os-disk-size-gb 64 --no-wait
+```
+
+## Enjoy your cluster
+
+You should now be able to interact with your cluster with `osctl`:
+
+```bash
+osctl --talosconfig ./talosconfig kubeconfig > kubeconfig
+kubectl --kubeconfig ./kubeconfig get nodes
+```
+
+You will need to apply a PSP and CNI configuration. More details can be found
+in the `getting started` guide.
```

Comment:
May want an actual link to the getting started guide here. Other than that, this rocks!